content/swiftui/concepts/notifications/notifications.md

```diff
@@ -0,0 +1,190 @@
+
+>> `ToastView`: A custom view that represents the toast. It contains a Text view displaying the provided message, with a dark background and white text color. The .frame, .padding, and .animation modifiers are used to control the appearance and animation of the toast.
+
+*This code will show a short confirmation message after the user successfully saves some data.*
+
+## 4. Custom Notifications 
+
+We can create our own custom notifications by designing custom views and animations. These can be used to provide unique and engaging user interface elements tailored to your app's style and branding.
+
+Custom notifications allow us to get creative with how we present information or engage users within our SwiftUI app.
+
+The specific implementation and appearance of notifications can be customized based on our app's design and user experience goals. SwiftUI's flexible and declarative approach makes it easy to create engaging and user-friendly notifications that enhance your app's overall experience.
+
+```struct ContentView: View {
+    var body: some View {
+        VStack {
+            Text("Welcome to My App")
+                .font(.title)
+            
+            Button("Show Custom Notification") {
+                // Trigger custom notification display
+            }
+            .padding()
+        }
+    }
+}
+```
+
+>> The code was left more open-ended to illustrate the concept of creating custom notifications. 
+The focus is on the idea that you can design and implement your own custom views and animations to create unique and visually appealing notifications tailored to your app's style and branding.
+
+
+
+
+## Best Practices
+
++ **Descriptive Names:**
+Use clear and descriptive names for your notifications to convey their purpose effectively. Consider using enums to define notification names for type safety.
+
++ **Decoupling:** 
+Utilize notifications to decouple different app components, but avoid using them excessively for closely related components. Also, use more direct communication methods for hierarchical relationships.
+
++ **Observer Removal:**
+ Always remove observers when they are no longer needed to prevent memory leaks. Typically, remove observers in the deinit method of the observing class.
+
++ **Avoid Overuse:**
+ While powerful, notifications should be used judiciously. Overusing them can lead to code complexity and reduced maintainability. Consider other communication patterns for more appropriate scenarios.
+
++ **Combine Framework:**
+ If using SwiftUI, explore the Combine framework to handle notifications and asynchronous tasks in a more reactive and functional way. Combine provides publishers and subscribers for improved code readability and organization.
```

Comment:
```suggestion
- **Descriptive Names:** Use clear and descriptive names for your notifications to convey their purpose effectively. Consider using enums to define notification names for type safety.
- **Decoupling:** Utilize notifications to decouple different app components, but avoid using them excessively for closely related components. Also, use more direct communication methods for hierarchical relationships.
- **Observer Removal:** Always remove observers when they are no longer needed to prevent memory leaks. Typically, remove observers in the deinit method of the observing class.
- **Avoid Overuse:**  While powerful, notifications should be used judiciously. Overusing them can lead to code complexity and reduced maintainability. Consider other communication patterns for more appropriate scenarios.
- **Combine Framework:** If using SwiftUI, explore the Combine framework to handle notifications and asynchronous tasks in a more reactive and functional way. Combine provides publishers and subscribers for improved code readability and organization.
```